src/hardware/compressor.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include <algorithm>
+#include <cmath>
+
+#include "compressor.h"
+#include "mixer.h"
+
+constexpr float log_to_db = 8.685889638065035f;  // 20.0 / log(10.0)
+constexpr float db_to_log = 0.1151292546497022f; // log(10.0) / 20.0
+
+std::array<float, 120> attack_times_ms = {};
+
+constexpr void fill_attack_times_lut()
+{
+	for (size_t i = 0; i < attack_times_ms.size(); ++i) {
+		const auto n = i + 1;
+		attack_times_ms[i] = ((0.08924f / n) + (0.60755f / (n * n)) - 0.00006f);
+	}
+}
+
```

Comment:
If the compressor is working well to your ear in this current state, then maybe we can replace the LUT with just zeros?

Currently the the `fill_attack_times_lut()` isn't being called - so the LUT is returning zeros. Could be, we don't need the table?

Reply:
Something like this will populate it at compile time:
``` c++
using attack_times_lut_t = std::array<float, 120>;

constexpr attack_times_lut_t fill_attack_times_lut()
{
	int16_t n = 1;
	attack_times_lut_t lut = {};
	for (auto &time_ms : lut) {
		time_ms = ((0.08924f / n) + (0.60755f / (n * n)) - 0.00006f);
		++n;
	}
	return lut;
}

constexpr auto attack_times_ms = fill_attack_times_lut();
```

Reply:
Ooops, good catch! After I've done all these improvements, I just smoke-tested the compressor quickly to see that it still works, but haven't done another round of detailed checks. So I'll do that now; maybe using a fixed, non-level-dependent attack is all we need for our purposes.

Reply:
I'm just reinstating the variable attack time stuff for now using your method, then I'll do further testing with fixed attack times later.